utils/file-manager.js

```diff
@@ -288,71 +275,35 @@ export default class FileManager {
 
   /**
    * Return the file path(s) that match the ID.
-   * If multiple files are found and `unique` is set to true,
-   * print a warning and return only the first one.
-   * If no file is found, return null.
    * @param {String} id
-   * @param {Boolean} unique return a single result
-   * @returns a String (unique=true), an array (unique=false), or null
+   * @returns an array
    */
-  getFilePathsFromId(id, unique = true) {
+  getFilePathsFromId(id) {
     if (!(id in this.idToGlob)) {
-      return null
+      throw new Error('ID is not defined')
     }
     const glob = this.idToGlob[id]
-    const paths = this.findMatchingFilePaths(glob)
-    if (paths.length === 0) {
-      return null
-    } else if (paths.length > 1) {
-      if (unique) {
-        console.warn(`Multiple files were found for id="${id}", glob="${glob}"`)
-      } else {
-        return paths
-      }
-    }
-    return paths[0]
+    return this.findMatchingFilePaths(glob)
   }
 
   /**
    * Return the preprocessed text of the file(s) that match the ID.
-   * If multiple files are found and `unique` is set to true,
-   * print a warning and return only the first one.
-   * If no file is found, return null.
    * @param {String} id
-   * @param {Boolean} unique return a single result
-   * @returns a String (unique=true), an array (unique=false), or null
+   * @returns an array
    */
-  async getPreprocessedTextFromId(id, unique = true) {
-    const paths = this.getFilePathsFromId(id, unique)
-    if (paths === null) {
-      return null
-    }
-    if (unique) {
-      return await this.getPreprocessedText(paths)
-    } else {
-      return await Promise.all(paths.map(p => this.getPreprocessedText(p)))
-    }
+  async getPreprocessedTextFromId(id) {
```

Comment:
You no longer need the second argument? That's an improvement I didn't expect.

Reply:
I think you may have missed my other comment https://github.com/hestiaAI/hestialabs-experiences/pull/523#discussion_r791723890